GitCommands/RevisionReader.cs

```diff
@@ -37,23 +40,33 @@ public sealed class RevisionReader
         private const int _offsetDaysForOldestBody = 6 * 30; // about 6 months
 
         // reflog selector to identify stashes
-        private readonly bool _hasReflogSelector;
+        private bool _hasReflogSelector;
 
-        private string LogFormat => _hasReflogSelector ? FullFormat.Replace("%B", "%gD%n%B") : FullFormat;
+        // Buffer to decode subject
+        private char[] _decodeBuffer = new char[4096];
 
-        public RevisionReader(GitModule module, bool hasReflogSelector, bool allBodies = false)
-            : this(module, hasReflogSelector, module.LogOutputEncoding, allBodies ? 0 : GetUnixTimeForOffset(_offsetDaysForOldestBody))
+        public RevisionReader(GitModule module, bool allBodies = false)
+            : this(module, module.LogOutputEncoding, allBodies ? 0 : GetUnixTimeForOffset(_offsetDaysForOldestBody))
         {
         }
 
-        private RevisionReader(GitModule module, bool hasReflogSelector, Encoding logOutputEncoding, long oldestBody)
+        private RevisionReader(GitModule module, Encoding logOutputEncoding, long oldestBody)
         {
             _module = module;
-            _hasReflogSelector = hasReflogSelector;
             _logOutputEncoding = logOutputEncoding;
             _oldestBody = oldestBody;
         }
 
+        /// <summary>
+        /// Return the git-log format and set the reflogSelector flag
+        /// (used when parsing the buffer).
+        /// </summary>
+        private string LogFormat(bool hasReflogSelector = false)
+        {
+            _hasReflogSelector = hasReflogSelector;
```

Comment:
Currently `RevisionReader` is stateless and side-effect free - i.e., for a given git repo it could be reused over and over; with this change we won't be able to do that.

Reply:
It is not stateless, the reflog selector must match the Git command.
This improves the setup as the user is not involved in setting the value.
(Before you created an instance with the setting that had to match the command.)

You cannot launch multiple parsers with the same reader instance, before or now.